karrot/utils/image_utils.py

```diff
@@ -0,0 +1,29 @@
+from io import BytesIO
+from typing import Tuple
+
+from PIL import Image
+from PIL.ImageOps import exif_transpose
+from django.core.files import File
+
+# A list of mime types that pillow supports
+Image.init()
+PILLOW_SUPPORTED_MIME_TYPES = {Image.MIME[id] for id in Image.ID if id in Image.MIME}
+
+
+def is_supported_content_type(content_type: str) -> bool:
+    if not content_type:
+        return False
+
+    return content_type.lower() in PILLOW_SUPPORTED_MIME_TYPES
+
+
+def resize_image(image: Image, size: Tuple[int, int]) -> File:
+    # processes rotation if present
+    image = exif_transpose(image)
+    # remove alpha if it has it
+    if image.mode in ["RGBA", "P"]:
+        image = image.convert('RGB')
+    image.thumbnail(size)
```

Comment:
yes I think it is, _but_ the exif_transpose step above (and the possible rgb conversion step) ensures we are working with a copy by that point:

https://pillow.readthedocs.io/en/stable/reference/ImageOps.html#PIL.ImageOps.exif_transpose says:

> If an image has an EXIF Orientation tag, other than 1, return a new image that is transposed accordingly. The new image will have the orientation data removed.
>
> Otherwise, return a copy of the image.